File: src/main/java/zwylair/zwym/entities/CubeEntityRenderer.java
package zwylair.zwym.entities;

import net.minecraft.client.render.entity.EntityRendererFactory;
import net.minecraft.client.render.entity.MobEntityRenderer;
import net.minecraft.util.Identifier;
import zwylair.zwym.EntityTestingClient;


/*
 * A renderer is used to provide an entity model, shadow size, and texture.
 */
public class CubeEntityRenderer extends MobEntityRenderer<CubeEntity, CubeEntityModel> {

    public CubeEntityRenderer(EntityRendererFactory.Context context) {
        super(context, new CubeEntityModel(context.getPart(EntityTestingClient.MODEL_CUBE_LAYER)), 0.5f);
    }

    @Override
    public Identifier getTexture(CubeEntity entity) {
        return new Identifier("zwym", "textures/entity/cube/cube.png");
    }
}

File: src/main/kotlin/zwylair/zwym/entities/CubeEntityRenderer.kt
package zwylair.zwym.entities

import net.minecraft.client.render.entity.EntityRendererFactory
import net.minecraft.client.render.entity.MobEntityRenderer
import net.minecraft.util.Identifier
import zwylair.zwym.ZwyM
import zwylair.zwym.ZwyMClient

/*
* A renderer is used to provide an entity model, shadow size, and texture.
*/
class CubeEntityRenderer(context: EntityRendererFactory.Context) :
    MobEntityRenderer<CubeEntity?, CubeEntityModel?>(
        context,
        CubeEntityModel(context.getPart(ZwyMClient.MODEL_CUBE_LAYER)),
        0.3f
    ) {
    override fun getTexture(entity: CubeEntity?): Identifier {
        return ZwyM.id("textures/entity/cube/cube.png")
    }
}
